### PR TITLE
Gareth/image urls

### PR DESCRIPTION
**Pull request by pcGarethJames**
Change Image URLs in product feed to be handled by a new class, which correctly sends a thumbnail image, rather than the full size.

### CODE REVIEW

File: Model/ProductExport.php [hunk 1]
```diff
@@ -350,7 +324,7 @@ public function processProduct(&$product, $index)
                 "InStock" => $this->stockRegistry->getStockItem($product->getId())->getIsInStock() ? 'true' : 'false'
             ];
 
-            if (sizeof($allImages) > 0) {
+            if (count($allImages) > 0) {
```
Comment (jcbarton) [inline Model/ProductExport.php]: why this change? Is it safe on all the versions of Magento we support?

File: Model/ProductExport/Images.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        $galleryBlock = $this->getGalleryBlock();
+        $galleryBlock->setData('product', $product);
+
+        $this->galleryReadHandler->execute($product);
+        $productImages = $galleryBlock->getGalleryImages()->getItems();
+
+        $allImages = [];
+
+        foreach ($productImages as $image) {
+            $allImages[] = str_replace(
```
Comment (jcbarton) [inline Model/ProductExport/Images.php]: do we potentially need to handle different cases? i.e. HTTP:// (given we're talking about arbitrary data)

File: Model/Feed.php [hunk 3]
```diff
@@ -203,6 +214,8 @@ public function sendProducts($pageSize = 50)
         }
 
         $this->logger->debug("PureClarity: Finished sending product data");
+
```
Comment (jcbarton) [inline Model/Feed.php]: try/catch to ensure we always end emulation regardless of what happens in the export?

File: Model/ProductExport.php (hunk 1 above)
Comment (pcGarethJames) [reply Model/ProductExport.php]: sizeof is discouraged in Magento 2 coding standards, so this removes a warning when running our code against phpcs (it's actually just an alias of count, and it's been suggested it might be deprecated in newer versions of PHP too)

Comment (pcGarethJames) [reply Model/ProductExport.php]: also, thought since I was changing how we generated $allImages, I may as well add this one in too :)

File: Model/ProductExport/Images.php (hunk 2 above)
Comment (pcGarethJames) [reply Model/ProductExport/Images.php]: Don't think we have to :), these URLs will only come from configuration and Magento doesn't allow you to put "HTTP://" as it treats that as an invalid URL. So in theory, we should never get an uppercase protocol in these URLs.
